### PR TITLE
feat: bulk update records using UI

### PR DESCRIPTION
**Pull request by mertmit**
## Change Summary

Re #5045
- This PR allows bulk update on both selected records or all records in a view.
  - Select records (or all records in case you want to update all in current view) then right click

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)

image
image
image
image

### CODE REVIEW

File: packages/nc-gui/components/dlg/BulkUpdate.vue [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+        <div v-if="updateMode === BulkUpdateMode.ALL" class="flex items-center">
+          <component :is="iconMap.contentSaveExit" class="mr-1" />
+          Bulk Update All
+        </div>
+        <div v-if="updateMode === BulkUpdateMode.SELECTED" class="flex items-center">
+          <component :is="iconMap.contentSaveStay" class="mr-1" />
+          Bulk Update Selected
+        </div>
+      </a-dropdown-button>
+      -->
```
Comment (wingkwong) [inline packages/nc-gui/components/dlg/BulkUpdate.vue:219]: if these are not in use, suggest to remove.

File: packages/nc-gui/components/dlg/BulkUpdate.vue [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+      <a-button v-if="props.selectedAllRecords" class="nc-bulk-update-save-btn" type="primary" @click="save">
+        <div class="flex items-center">
+          <component :is="iconMap.contentSaveExit" class="mr-1" />
+          Bulk Update All
```
Comment (wingkwong) [inline packages/nc-gui/components/dlg/BulkUpdate.vue:223]: i18n or add a TODO tag. apply to other texts.

**Review by dstala — CHANGES_REQUESTED** [review]
- [x] [**Marked by Naveen as not supported**] Sakila ext DB PG table. Actor table. Select all - bulk update. Unable to `undo`. Where as Undo works for smaller subset of bulk update.
- [ ] Good to show all the columns in bulk update modal? I would prefer to see only visible fields (not hidden). Discuss.
- [x] UI background needs to be sorted out when we have many columns to update in bulk modal
Screenshot 2023-06-20 at 11 27 40 AM

- [x] Bulk update when a filter is enabled threw error (issue not noticed in secondary view)
Screenshot 2023-06-20 at 11 29 53 AM

**Review by dstala — APPROVED** [review]

**Comment by dstala** [comment]
- [ ] Long text : let us display bigger block on loading it into bulk upload form instead of toggling to bigger block (text area) on click. This way- clicking on next fields after updating long text doesn't realign blocks
- [ ] Tab doesn't work when navigating between fields in upload form
- [ ] Checkbox, on first load is not aligned (somehow the square box appears at the bottom of circle). I am using chrome.
  
Screenshot 2023-06-23 at 5 07 39 PM

- [ ] Date picker (or time/year picker) : after opening it - escape key should only close picker. Subsequent escape should close Bulk update modal
- [ ] Let us swap close icon with hide icon (eye) in form columns. Hide icon should appear only on hovering over that particular column; otherwise it remains hidden (as in form view). Currently form view hide icon is tad bigger. You can keep reduce its size a bit.
- [ ] Consider removing border (my opinion- ignore if you do not agree) to make it look similar to expanded form.
Screenshot 2023-06-23 at 5 11 39 PM

**Pull request merged**